### DISCUSSION TITLE
Manually run YOLOv8

### DISCUSSION THREAD

**Discussion by laminarize** [ultralytics/ultralytics#2629] [Q&A]
Hey everyone - I wanted to manually run YOLOv8 by using CLI and python "some_python_file.py"

My goal is to trace the dataflow through the different python files as I run a live inference and I want to do that using viztracer. Any suggestions?

**Comment by glenn-jocher** [comment]
@laminarize hello! Running YOLOv8 manually for live inference and tracing the dataflow is a great way to understand the inner workings of the model. Here's a high-level approach to achieve this:

1. **CLI Approach**: You can use the CLI to run predictions with YOLOv8. Start by setting up your environment and then use the `predict` mode with the appropriate flags for your live inference. This will allow you to see the output directly in your terminal.

2. **Python Script**: For a more detailed trace, write a Python script that loads the YOLOv8 model and processes frames for live inference. You can then use `viztracer` to run your script and generate the trace report. Make sure to import the necessary modules and initialize the model correctly in your script.

3. **Tracing**: With `viztracer`, you can start the tracing before the inference loop and stop it after you've processed the desired number of frames or time duration. This will capture the dataflow through the different Python files during live inference.

Remember to review the documentation for detailed explanations of each mode and how to use them effectively. Happy tracing! 😊🔍
